pkg/liqoctl/offload/handler.go

```diff
@@ -72,6 +77,63 @@ func (o *Options) Run(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctx, o.Timeout)
 	defer cancel()
 
+	var offloadNamespaces []string
+	offloadNamespaces = append(offloadNamespaces, o.Namespaces...)
+
+	if o.LabelSelector != "" {
+		// Parse the label selector
+		selector, err := labels.Parse(o.LabelSelector)
+		if err != nil && o.LabelSelector != "" {
+			return fmt.Errorf("invalid label selector: %w", err)
+		}
+
+		// List namespaces
+		var selectedNsList corev1.NamespaceList
+		if err := o.CRClient.List(ctx, &selectedNsList, &client.ListOptions{LabelSelector: selector}); err != nil {
+			return fmt.Errorf("cannot list namespace objects: %w", err)
+		}
+
+		for _, ns := range selectedNsList.Items {
+			if !slices.Contains(offloadNamespaces, ns.Name) {
+				offloadNamespaces = append(offloadNamespaces, ns.Name)
+			}
+		}
+	}
+
+	if len(offloadNamespaces) == 0 {
+		o.Printer.Info.Println("No namespaces can be offloaded")
+		return nil
+	}
+
+	// Offload each namespace
+	var errors []error
+	for _, ns := range offloadNamespaces {
+		// Create a new options instance for each namespace to avoid state pollution
+		nsOptions := &Options{
+			Factory:                  o.Factory,
+			Namespace:                ns,
+			PodOffloadingStrategy:    o.PodOffloadingStrategy,
+			NamespaceMappingStrategy: o.NamespaceMappingStrategy,
+			RemoteNamespaceName:      o.RemoteNamespaceName,
+			ClusterSelector:          o.ClusterSelector,
+			OutputFormat:             o.OutputFormat,
+			Timeout:                  o.Timeout,
+		}
+
+		if err := nsOptions.runOffload(ctx); err != nil {
+			errors = append(errors, err)
+		}
```

Comment:
I couldn't understand why you create a new option struct for each namespace you need to offload.

I would just call an internal function where we provide the namespace name, it seems more straightforward, and there is less risk to forget passing some of the properties of the option struct:
```suggestion
		if err := o.runOffload(ctx, ns); err != nil {
			errors = append(errors, err)
		}
```